cli_grader/src/grader/grading_tests/unit_test.rs

```diff
@@ -1,50 +1,73 @@
-use crate::grader::assessment_modalities::unit_test::assertion::Assertion;
-use crate::grader::score::{Mode, Score};
+use crate::grader::grading_tests::unit_test::assertion::Assertion;
+use crate::grader::score::{GradingMode, Score};
 
-pub mod assertion;
+pub(crate) mod assertion;
 
-use crate::grader::executable::ExecutableArtifact;
+use crate::input::ExecutableArtifact;
 use assertion::AssertionResult;
 use std::{fs, io, process};
 
 #[derive(Debug, PartialEq, Eq, Clone)]
-pub struct AssertionsPerExecutable {
+pub struct UnitTest {
     name: String, // Default: `Unit testing <TargetProgram>`
+    // TODO does it make sense to allow multiple executables to be tested under the same set
+    // of assertions?
```

Comment:
(create an issue) Yes. One use case I can think of is testing multiple versions of a program for the same test suite.